js/ui/components/networkAgent.js

```diff
@@ -151,6 +168,16 @@ const NetworkSecretDialog = new Lang.Class({
         this._updateOkButton();
     },
 
+    _addKbdLayoutSwitcher: function() {
+        this._inputSourceIndicator = new InputSourceIndicator(this, false);
+        this._manager = new PopupMenu.PopupMenuManager({ actor: this._inputSourceIndicator.container });
+        this._manager.addMenu(this._inputSourceIndicator.menu);
+        this._inputSourceIndicator.connect('destroy', Lang.bind(this, function() {
```

Comment:
I don't think this is really necessary -- I'd expect the container to also be destroyed when the parent actor is destroyed... The real issue here I can see is that we're "abusing" a bit the container/panel menu hierarchy, but that's probably fine...